fbw-a380x/src/systems/instruments/src/EWD/elements/WdLimitations.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+  ClockEvents,
+  ConsumerSubject,
+  DisplayComponent,
+  EventBus,
+  FSComponent,
+  Subject,
+  VNode,
+} from '@microsoft/msfs-sdk';
+import { FormattedFwcText } from 'instruments/src/EWD/elements/FormattedFwcText';
+import { EwdSimvars } from 'instruments/src/EWD/shared/EwdSimvarPublisher';
+import EWDMessages from '@instruments/common/EWDMessages';
+
+interface WdLimitationsProps {
+  bus: EventBus;
+}
+
+const padEWDCode = (code: number) => code.toString().padStart(9, '0');
+
+export class WdLimitations extends DisplayComponent<WdLimitationsProps> {
+  private readonly sub = this.props.bus.getSubscriber<ClockEvents & EwdSimvars>();
+
+  private readonly limitationsLeftSvgRef = FSComponent.createRef<SVGGraphicsElement>();
+
+  private readonly limitationsRightSvgRef = FSComponent.createRef<SVGGraphicsElement>();
+
+  private readonly limitationsLeft = Array.from(Array(10), (val, idx) =>
+    ConsumerSubject.create(this.sub.on(`limitations_all_${idx + 1}`).whenChanged(), 0),
+  );
+
+  private readonly limitationsRight = Array.from(Array(10), (val, idx) =>
+    ConsumerSubject.create(this.sub.on(`limitations_apprldg_${idx + 1}`).whenChanged(), 0),
+  );
+
+  private readonly limitationsLeftFormatString = Subject.create('');
+
+  private readonly limitationsRightFormatString = Subject.create('');
+
+  private readonly limitationsDisplay = Subject.create('none');
+
+  public onAfterRender(node: VNode): void {
+    super.onAfterRender(node);
+
+    this.sub
+      .on('realTime')
```

Comment:
Any reason to not use the subs from `limitationsLeft` and `limitationsRight` to set the strings? Then we would avoid running the logic if nothing changed, or?

Reply:
True. Will push a better version with the next commit